cmd/libs/go2idl/client-gen/generators/fake/generator_fake_for_type.go

```diff
@@ -78,16 +79,31 @@ func (g *genFakeForType) GenerateType(c *generator.Context, t *types.Type, w io.
 	pkg := filepath.Base(t.Name.Package)
 	const pkgTestingCore = "k8s.io/kubernetes/pkg/client/testing/core"
 	namespaced := !(types.ExtractCommentTags("+", t.SecondClosestCommentLines)["nonNamespaced"] == "true")
+	canonizeGroup := func(group string) string {
+		if group == "core" {
+			return ""
+		}
+		return group
+	}
+	canonizeVersion := func(version string) string {
+		if version == "unversioned" {
+			return ""
+		}
+		return version
+	}
 	m := map[string]interface{}{
-		"type":             t,
-		"package":          pkg,
-		"Package":          namer.IC(pkg),
-		"namespaced":       namespaced,
-		"Group":            namer.IC(g.group),
-		"watchInterface":   c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/watch", Name: "Interface"}),
-		"apiDeleteOptions": c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/api", Name: "DeleteOptions"}),
-		"apiListOptions":   c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/api", Name: "ListOptions"}),
-		"Everything":       c.Universe.Function(types.Name{Package: "k8s.io/kubernetes/pkg/labels", Name: "Everything"}),
+		"type":                 t,
+		"package":              pkg,
+		"Package":              namer.IC(pkg),
+		"namespaced":           namespaced,
+		"Group":                namer.IC(g.group),
+		"group":                canonizeGroup(g.group),
```

Comment:
Why did you write anonymous functions if it's only going to be called once. Why not just use a variable?

```go
canonicalGroup := g.group
if canonicalGroup == "core" {
	canonicalGroup = ""
}

...

"group": canonicalGroup
```